### PR TITLE
Support plugin event

### PR DESCRIPTION
**Pull request by ben181231**
connect #100

### CODE REVIEW

**Comment by oursky-travis** [comment]
@ben181231, thanks for your PR! By analyzing the annotation information on this pull request, we identified @rickmak to be a potential reviewer

File: lib/cloud/registry.js [hunk 1]
```diff
@@ -72,6 +74,16 @@ export class Registry {
...
   }
 
+  registerEvent(name, func, options) {
+    const opts = {
+      ...options,
+      name
+    };
+
+    this._addParam('event', opts);
+    this.funcMap.event[name] = func;
```
Comment (rickmak) [inline lib/cloud/registry.js]: We will need to support multiple func hook on to one event. So the name cannot just set according to the event name.

File: lib/cloud/transport/http.js [hunk 2]
```diff
@@ -42,9 +43,17 @@ class HTTPTransport extends CommonTransport {
     this.server.on('error', (err, socket) => {
       socket.end('HTTP/1.1 500 Internal Server Error\r\n\r\n');
     });
-    console.info('Listening on: ' + settings.httpAddr);
-    // TODO: parse the httpAddr
-    this.server.listen(9000, '0.0.0.0');
+
```
Comment (rickmak) [inline lib/cloud/transport/http.js]: This split should belong to setting modules.

Comment (ben181231) [inline lib/cloud/transport/http.js]: Okay

File: lib/cloud/registry.js (hunk 1 above)
Comment (ben181231) [inline lib/cloud/registry.js]: Oh...

File: Dockerfile-development [hunk 3]
```diff
@@ -5,11 +5,20 @@ RUN \
...
+
+COPY . /usr/src/skygear
 
-COPY . /usr/src/app
-RUN (cd /usr/src/app; npm install && npm link)
+RUN \
+  cd /usr/src/skygear && \
+  npm install && \
+  npm run prepublish && \
+  npm install -g $PWD
```
Comment (cheungpat) [inline Dockerfile-development]: This Dockerfile is aimed for skygear contributor, and the previous code allow development without rebuilding Docker image every step of the way

Comment (ben181231) [inline Dockerfile-development]: I got that.

**Comment by ben181231** [comment]
Updated

**Comment by cheungpat** [comment]
@rickmak please merge this for release

**Pull request merged**